samples/bluetooth/peripheral_uart/src/main.c

```diff
@@ -206,6 +220,14 @@ static void uart_work_handler(struct k_work *item)
 	uart_rx_enable(uart, buf->data, sizeof(buf->data), UART_WAIT_FOR_RX);
 }
 
+static bool uart_test_async_api(const struct device *dev)
+{
+	const struct uart_driver_api *api =
+			(const struct uart_driver_api *)dev->api;
+
+	return !!(api->callback_set);
```

Comment:
I like this form because it is short. But I will use api->callback != NULL worrying if anybody else would note it as error and wish to "fix" it. Basically:

! - negate
!! - convert to bool

Reply:
It would be better. It is easy to take is as a mistake now

Reply:
fixed